### PR TITLE
Major update of this role

### PR DESCRIPTION
**Pull request by haslersn**

### CODE REVIEW

File: templates/dhcpd.conf.j2 [hunk 1]
```diff
@@ -1,110 +1,31 @@
...
-  load balance max seconds {{ isc_dhcp_server_failover.load_balance_max_seconds }};
-{% endif %}
-{% if isc_dhcp_server_failover.max_lease_misbalance is defined %}
-  max-lease-misbalance {{ isc_dhcp_server_failover.max_lease_misbalance }};
-{% endif %}
-{% if isc_dhcp_server_failover.max_lease_ownership is defined %}
-  max-lease-ownership {{ isc_dhcp_server_failover.max_lease_ownership }};
+{% for subnet, options in isc_dhcp_server_subnets.items() %}
+{% if subnet | ipv4 %}
+subnet {{ subnet | ipaddr('network') }} netmask {{ subnet | ipaddr('netmask') }} {
```
Comment (neumantm) [inline templates/dhcpd.conf.j2]: Maybe wo should use `ansible.netcommon.ipaddr`? See https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html

File: templates/dhcpd.conf.j2 [hunk 2]
```diff
@@ -1,110 +1,31 @@
...
+{% if options | map('regex_search', '^range ') | select('string') | length == 0 %}
+  range {{ subnet | ipaddr('100') | ipaddr('address') }} {{ subnet | ipaddr('-2') | ipaddr('address') }};
```
Comment (neumantm) [inline templates/dhcpd.conf.j2]:
```suggestion
  range {{ subnet | ansible.netcommon.ipmath('100') | ipaddr('address') }} {{ subnet | ansible.netcommon.ipmath('-2') | ipaddr('address') }};
```
would increase readability.

Comment (neumantm) [inline templates/dhcpd.conf.j2]: What if the user specifies a subnet with less then 100 IPs?
What would we expect to happen/what do we want to happen?
What happens with config generated by this template?

**Review by neumantm — CHANGES_REQUESTED** [review]

File: templates/dhcpd.conf.j2 (hunk 2 above)
Comment (haslersn) [reply templates/dhcpd.conf.j2]: `ipmath('-2')` subtracts 2 whereas `ipaddr('-2')` returns the 2nd-to-last IP address.

File: templates/dhcpd.conf.j2 (hunk 1 above)
Comment (haslersn) [reply templates/dhcpd.conf.j2]: Yes, I added a commit.

File: templates/dhcpd.conf.j2 (hunk 2 above)
Comment (haslersn) [reply templates/dhcpd.conf.j2]: I added a check in my recent commit.

Comment (neumantm) [reply templates/dhcpd.conf.j2]: Maybe `nthhost` would work?

File: README.md [hunk 3]
```diff
@@ -1,46 +1,79 @@
...
+| `isc_dhcp_server_subnets`     |        `{}`        | Dict of [subnet declarations](#subnet-declarations)                                      |
+| `isc_dhcp_server_hosts`       |        `{}`        | Dict of [host declarations](#host-declarations)                                          |
+
+### Subnet Declarations
+
+In a subnet declaration, the key is the router IP followed by the netmask in CIDR notation.
+The value is a list of configuration statements (excluding semicolon) for that subnet.
+There are certain defaults:
+
+- If no `range` statement is given, then a range from network address plus 100 to broadcast address minus one will be configured.
```
Comment (neumantm) [inline README.md]: Document the condition of prefix <= 25

File: templates/dhcpd.conf.j2 (hunk 2 above)
Comment (haslersn) [reply templates/dhcpd.conf.j2]: That works, I updated it.

File: README.md (hunk 3 above)
Comment (haslersn) [reply README.md]: Done

**Review by neumantm — APPROVED** [review]

**Pull request merged**